cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -458,6 +477,16 @@ void BuildStringDictionaryIndex(ColumnChunkDesc* chunks,
                                 int32_t num_chunks,
                                 rmm::cuda_stream_view stream);
 
+/**
+ * @brief Get OR'd sum of page kernel masks.
+ *
+ * @param[in] pages List of pages to aggregate
+ * @param[in] stream CUDA stream to use
+ * @return Bitwise OR of all page `kernel_mask` values
+ */
+uint32_t SumPageKernelMasks(cudf::detail::hostdevice_vector<PageInfo>& pages,
```

Comment:
This function name feels a little oblique.  I think `GetKernelMask` would be cleaner, with a description like "Determine the set of kernels that need to be invoked on these pages."

Reply:
My concern with `GetKernelMask` is that it looks like a getter. If it were something like `GetAggregatedKernelMask` it would be more clear to me that some reduction is happening in the functions.

Reply:
Also coming down the pike is encode kernel masks...how about `GetAggregateDecodeKernelMasks`?

Reply:
Yeah, that sounds good to me.